app/src/routes/api/v1/dataset.router.js

```diff
@@ -260,6 +261,20 @@ class DatasetRouter {
         }
 
         try {
+            if (query.sort && (query.sort.includes('user.role') || query.sort.includes('user.name'))) {
+                logger.debug('Detected sorting by user role or name');
+                if (!user || !['ADMIN', 'SUPERADMIN'].includes(user && user.role)) {
```

Comment:
the fact that this works is why I actually hate JS as a language

Reply:
Extracted `isAdmin` check to a variable, not much left to do here 🤷‍♂